beamer/agent.py

```diff
@@ -115,6 +118,7 @@ def stop(self) -> None:
         self._event_monitor_l2b.stop()
         self._task_pool.shutdown(wait=True, cancel_futures=False)
         self._stopped.set()
+        self._init()
```

Comment:
I would put the call to `_init` one line above, before `self._stopped.set()` so that whoever calls `wait()` from a different thread does not get to run before we reinitialize the agent instance.